src/inmanta/agent/handler.py

```diff
@@ -551,28 +551,49 @@ def _should_reload() -> bool:
                 raise Exception(f"Failed to determine whether resource should reload{error_msg_from_server}")
             return result.result["data"]
 
-        def filter_resources_in_unexpected_state(
-            reqs: abc.Mapping[ResourceIdStr, ResourceState]
+        def filter_resources_by_state(
+            reqs: abc.Mapping[ResourceIdStr, ResourceState], states: typing.Set[const.ResourceState]
```

Comment:
Let's make the typing consistent with the `reqs` parameter.
```suggestion
            reqs: abc.Mapping[ResourceIdStr, ResourceState], states: typing.Set[ResourceState]
```